gitpod-io/gitpod
# [Tests] Add fallback logic when Workspace URL is empty

## Description
This condition often occurs when we run tests for Self-Hosted clusters, see example: [[gitpod-k3s-installer-tests-main.143]](https://werft.gitpod-dev.com/job/gitpod-k3s-installer-tests-main.143/logs).

By looking at where the Workspace URL is returned, we've added fallback logic to reduce the possibility for this to happen.

## Related Issue(s)

Fixes #14101 

## How to test
It was tested in https://werft.gitpod-dev.com/job/gitpod-custom-afalz-integration-tests-retry-workspace-urla.0 - VS Code failed for a timeout but both JB and SSH Gateway tests passed (testing this change x7 times).

To test again (it takes ~1 hour to test):
1. start a workspace for this branch
2. run: `werft run github -s .werft/installer-tests.ts -j .werft/k3s-installer-tests.yaml -a channel=unstable -a testSuite=ide`
3. check in Werft that all tests pass (it will take at least 1 hourr)

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
